Python/thread_nt.h

```diff
@@ -207,29 +207,59 @@ PyThread_start_new_thread(void (*func)(void *), void *arg)
         /* I've seen errno == EAGAIN here, which means "there are
          * too many threads".
          */
-        int e = errno;
-        threadID = (unsigned)-1;
         HeapFree(GetProcessHeap(), 0, obj);
+        return -1;
     }
-    else {
-        CloseHandle(hThread);
+    *ident = threadID;
+    *handle = (Py_uintptr_t) hThread;
```

Comment:
Would it make sense to do a quick assert before the cast here, like we do in `PyThread_start_joinable_thread()`:

```suggestion
    *handle = (Py_uintptr_t) hThread;
    assert(hThread == (HANDLE) *handle);
```

It we don't do an assert, at least we could have a comment indicating why we're confident there's no inconsistency (e.g. overflow).  I expect most readers would be the same as me: clueless about type compatibility on Windows.